server/src/main/java/org/elasticsearch/search/fetch/subphase/MatchedQueriesPhase.java

```diff
@@ -38,13 +38,11 @@
 public final class MatchedQueriesPhase implements FetchSubPhase {
 
     @Override
-    public FetchSubPhaseProcessor getProcessor(SearchContext context) throws IOException {
-        if (context.docIdsToLoadSize() == 0 ||
-            // in case the request has only suggest, parsed query is null
-            context.parsedQuery() == null) {
-            return null;
+    public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOException {
+        Map<String, Query> namedQueries = new HashMap<>();
```

Comment:
Checking I understand, does this actually fix a bug where we wouldn't return named query information if there was only a post filter?

Reply:
I don't think it's actually possible to have a post filter without a query, but the logic of what gets set where is pretty hairy so I thought it best to be extra defensive here.